supabase/migrations/20241127203353_searchable_index_col.sql

```diff
@@ -0,0 +1,36 @@
+ALTER TABLE topics
+    ADD COLUMN searchable_index_col tsvector;
+
+CREATE INDEX search_idx ON topics USING gin(searchable_index_col);
+
+CREATE OR REPLACE FUNCTION private.update_searchable_index_col()
+    RETURNS TRIGGER
+    AS $$
+DECLARE
+    message_content text;
+BEGIN
+    SELECT
+        m.content INTO message_content
+    FROM
+        public.messages m
+    WHERE
+        m.id = NEW.first_message_id;
+    NEW.searchable_index_col = to_tsvector('english', NEW.name || ' ' || message_content);
```

Comment:
Could `message_content` or `first_message_id` potentially be `null`?

Reply:
Yes, the first_message_id can be null.